include/swift/AST/DiagnosticsParse.def

```diff
@@ -208,8 +208,8 @@ ERROR(decl_already_static,none,
 ERROR(enum_case_dot_prefix,none,
       "extraneous '.' in enum 'case' declaration", ())
 
-ERROR(associatedtype_where_swift_3,none,
-      "where clauses on associated types are fragile; use '-swift-version 4' to experiment.", ())
+ERROR(protocol_associatedtype_where_swift_3,none,
+      "where clauses on %0 are fragile; use '-swift-version 4' to experiment.", (StringRef))
```

Comment:
Just easy to break and not completely implemented. (E.g. the compiler will likely (hopefully) just crash during IRGen because it isn't doing witness tables correctly yet.)